src/main/webapp/app/exercises/shared/result/result-detail.component.ts

```diff
@@ -449,4 +457,80 @@ export class ResultDetailComponent implements OnInit {
         const programmingSubmission = this.result.submission as ProgrammingSubmission;
         return createCommitUrl(this.commitHashURLTemplate, projectKey, this.result.participation, programmingSubmission);
     }
+
+    /**
+     * Updates the datasets of the charts with the correct values and colors.
+     * @param receivedPositive Sum of positive credits of the score
+     * @param appliedNegative Sum of applied negative credits
+     * @param receivedNegative Sum of received negative credits
+     * @param maxScore The relevant maximal points of the exercise
+     * @param maxScoreWithBonus The actual received points + optional bonus points
+     */
+    setValues(receivedPositive: number, appliedNegative: number, receivedNegative: number, maxScore: number, maxScoreWithBonus: number): void {
+        this.ngxData = [
+            {
+                name: 'Score',
+                series: [
+                    { name: this.labels[0], value: 0 },
+                    { name: this.labels[1], value: 0 },
+                ],
+            },
+        ];
+        let appliedPositive = receivedPositive;
+
+        // cap to min and max values while maintaining correct negative points
+        if (appliedPositive - appliedNegative > maxScoreWithBonus) {
+            appliedPositive = maxScoreWithBonus;
+            appliedNegative = 0;
+        } else if (appliedPositive > maxScoreWithBonus) {
+            appliedNegative -= appliedPositive - maxScoreWithBonus;
+            appliedPositive = maxScoreWithBonus;
+        } else if (appliedPositive - appliedNegative < 0) {
+            appliedNegative = appliedPositive;
+        }
+        const score = this.roundToDecimals(((appliedPositive - appliedNegative) / maxScore) * 100, 2);
+        this.xScaleMax = this.xScaleMax > score ? this.xScaleMax : score;
```

Comment:
We could use ``Math.max(a, b)`` here I guess?

Reply:
Thanks for this hint, I will incorporate it!